upgrade/steps_helpers.go

```diff
@@ -671,46 +671,103 @@ var getExpectedTargetFromIssues = stepv2.Func11E("From Issues", func(ctx context
 	}, nil
 })
 
+const pulumiupgradeproviderissue = "pulumiupgradeproviderissue"
+
 // Create an issue in the provider repo that signals an upgrade
 var createUpstreamUpgradeIssue = stepv2.Func30E("Ensure Upstream Issue", func(ctx context.Context,
 	repoOrg, repoName, version string) error {
 	upstreamProviderName := GetContext(ctx).UpstreamProviderName
 	upstreamOrg := GetContext(ctx).UpstreamProviderOrg
 	title := fmt.Sprintf("Upgrade %s to v%s", upstreamProviderName, version)
 
-	searchIssues := stepv2.Cmd(ctx, "gh", "search", "issues",
-		title,
+	issueAlreadyExists, err := upgradeIssueExits(ctx, title, repoOrg, repoName)
+	if err != nil {
+		return err
+	}
+
+	// Write issue_created=true to GITHUB_OUTPUT, if it exists for CI control flow.
+	if GITHUB_OUTPUT, found := os.LookupEnv("GITHUB_OUTPUT"); found {
+		f, err := os.OpenFile(GITHUB_OUTPUT, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0644)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		if _, err := f.WriteString(fmt.Sprintf("issue_created=%t\n", issueAlreadyExists)); err != nil {
+			return err
+		}
+	}
+
+	// We've found an appropriate existing issue, so we'll skip creating a new one.
+	if issueAlreadyExists {
+		return nil
+	}
+
+	// Hide searchable token in the issue body via an HTML comment to help us find this issue later without requiring labels to be set up.
+	hiddenBody := fmt.Sprintf("<!-- %s -->", pulumiupgradeproviderissue)
```

Comment:
Lets add a comment (visible to people who look at the opened issue description) that explains why this comment exists.